apps/page_mapping/src/main.c

```diff
@@ -186,17 +206,28 @@ bench_proc(int argc UNUSED, char *argv[])
     COMPILER_MEMORY_FENCE();
     send_result(result_ep, end - start);
 
-    /* cleaning */
+    /* Cleaning up the pages */
     long err;
     for (int i = 0; i < npage; i++) {
         err = seL4_ARCH_Page_Unmap(page_ptr_start + i);
         ZF_LOGF_IFERR(err, "ummap page failed\n");
     }
 
-    for (int i = 0; i < NUM_PAGE_TABLE(npage); i++) {
-        err = seL4_ARCH_PageTable_Unmap(pt_ptr_start + i);
-        ZF_LOGF_IFERR(err, "ummap page table failed\n");
-
+    /* Cleaning up the page table structures */
+    for (int i = pt_ptr_start; i < page_ptr_start; i++) {
+#ifdef CONFIG_ARCH_X86_64
+        err = seL4_X86_PDPT_Unmap(i);
+        if (err) {
+            err = seL4_X86_PageDirectory_Unmap(i);
+            if (err) {
+                err = seL4_ARCH_PageTable_Unmap(i);
+                ZF_LOGF_IFERR(err, "ummap page table\n");
+            }
+        }
+#else
```

Comment:
better make this more specific
```
#elif defined(CONFIG_ARCH_AARCH64)
```

Reply:
So it could be this:
```suggestion
#ifdef CONFIG_ARCH_X86_64
        err = seL4_X86_PDPT_Unmap(i);
        if (!err) { continue; }
        err = seL4_X86_PageDirectory_Unmap(i);
        if (!err) { continue; }
#endif /* CONFIG_ARCH_X86_64 */
        err = seL4_ARCH_PageTable_Unmap(i);
        ZF_LOGF_IFERR(err, "ummap page table\n");
```

Reply:
That would work, though I'm not sure it helps much in terms of clarity.